src/samples/network/echo-client.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * @brief Network link status
+ *
+ * Network sample that monitors specified links. It checks if the link is
+ * up/down and its addresses.
+ * To see the usage help, -h or --help.
+ */
+
+#include <errno.h>
+#include <getopt.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+
+#include "soletta.h"
+#include "sol-buffer.h"
+#include "sol-socket.h"
+#include "sol-network.h"
+#include "sol-vector.h"
+#include "sol-util.h"
+
+static struct sol_socket *sock;
+static struct sol_network_link_addr address;
+
+static bool
+on_can_read(void *data, struct sol_socket *s)
+{
+    ssize_t r;
+    char buffer[1024] = { };
+    struct sol_network_link_addr addr;
+
+    r = sol_socket_recvmsg(s, buffer, 1024, &addr);
+    if (r < 0) {
+        fprintf(stderr, "ERROR: Failed in receiving the message\n");
+        goto err;
+    }
+
+    printf("Received: %s\n", buffer);
+    sol_quit_with_code(EXIT_SUCCESS);
+
+    return true;
+
+err:
+    sol_quit_with_code(EXIT_FAILURE);
+    return false;
+}
+
+static bool
+on_can_write(void *data, struct sol_socket *s)
+{
+    int r = -1;
+    const char *value = data;
+
+    r = sol_socket_sendmsg(sock, value, strlen(value), &address);
+    if (r < 0) {
+        fprintf(stderr, "ERROR: Could not send data\n");
+        goto end;
+    }
+
+    r = 0;
+
+end:
+    sol_socket_set_write_monitor(sock, false);
```

Comment:
What's the difference of returning `false` and disabling the write monitor? From the documentation I got the impression that they are equivalent?